### PR TITLE
Added build script to create, package & publish plugins

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/1741

### CODE REVIEW

**Review by avasquez614 — APPROVED** [review]

File: resources/template/plugin.xml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  
+  <price>
+    <paid></paid> <!-- boolean, if not then it's free -->
+    <amount></amount>
+    <currency></currency>
+  </price>
+  
+  <crafter-versions-supported>
+    <range>
+      <min>4.0.0</min>
```
Comment (sumerjabri) [inline resources/template/plugin.xml]: 3.1.0 start
3.1.0 finish

File: resources/template/plugin.xml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  
+  <!-- Placeholders will be replaced during packaging -->
+  <build>
+    <date>@BUILD_DATE@</date>
+    <number>@BUILD_NUMBER@</number>
+  </build>
+  
+  <license>
+    <name></name>
+    <!-- Can be http:// or relative URL -->
```
Comment (sumerjabri) [inline resources/template/plugin.xml]: http(s)?

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,178 @@
+import java.nio.file.FileAlreadyExistsException
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.time.Instant
+
+import groovy.util.XmlParser
+
+import org.apache.tools.ant.filters.ReplaceTokens
+// import org.eclipse.egit.github.core.Issue
```
Comment (sumerjabri) [inline build.gradle:9]: Do we need this commented out code block?

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  }
+}
+
+task publishPlugin {
+  description "Publishes a plugin to the Crafter Marketplace"
+  
+  doLast {
+    if(autoPublish) {
+      println project.getProperty("publish.message.auto")
+      // ant.input(message: 'Pelase type your username', addproperty: 'username')
```
Comment (sumerjabri) [inline build.gradle]: `Please` (typo)

Also, shouldn't this indicate that you're asking for github creds?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

**Pull request merged**